internal/profile/profile.go

```diff
@@ -2,64 +2,87 @@ package profile
 
 import (
 	"fmt"
-	"strings"
 
 	"github.com/cyberark/secretless-broker/internal"
 	"github.com/pkg/profile"
 )
 
+// StartProfilingFunc is a function that will be invoked when profile start
+// is requested. This method signature is just a type wrapper around the
+// `Start()` signature from `github.com/pkg/profile`.
+type StartProfilingFunc func(...func(*profile.Profile)) interface{ Stop() }
+
+// Start is used to begin the profiling
 func (pp *perfProfile) Start() error {
-	switch pp.profileType {
-	case "cpu":
-		pp.stopper = profile.Start(profile.NoShutdownHook)
-	case "memory":
-		pp.stopper = profile.Start(profile.MemProfile, profile.NoShutdownHook)
-	default:
-		// will be impossible when New is used as ctor
-		return fmt.Errorf("attempt to start profiling with invalid profileType")
-	}
+	pp.stopper = pp.startProfilingFunc(pp.profileOptions...)
 	return nil
 }
 
+// Start is used to end the profiling
 func (pp *perfProfile) Stop() error {
 	pp.stopper.Stop()
 	return nil
 }
 
 type perfProfile struct {
-	profileType string
-	stopper interface { Stop() }
+	profileOptions     []func(*profile.Profile)
+	startProfilingFunc StartProfilingFunc
+	stopper            interface{ Stop() }
 }
 
-// ValidTypes are the valid types of profiling you can perform.
-var ValidTypes = []string{"cpu", "memory"}
+// validTypes are the valid types of profiling you can perform.
+var validTypes = []string{"cpu", "memory"}
 
 func isValidType(profileType string) bool {
-	for _, curType := range ValidTypes {
+	for _, curType := range validTypes {
 		if curType == profileType {
 			return true
 		}
 	}
 	return false
 }
 
-// ValidateType returns an error unless its argument is a valid profile type.
-func ValidateType(profileType string) error {
+func validateType(profileType string) error {
```

Comment:
This shoudn't be working, as we use this function in entrypoint.

Reply:
The change here makes that validation part of instantiating the `profile` service: https://github.com/cyberark/secretless-broker/pull/981/files#diff-06fef48cb3a3ba8ce687ec606fb1f771R82. There's no need for entrypoint to know what are the exact valid types since this class can do it on it's own and return an error otherwise.